internal/backend/remote-state/kubernetes/client.go

```diff
@@ -90,46 +120,94 @@ func (c *RemoteClient) Put(data []byte) error {
 		return err
 	}
 
-	secret, err := c.getSecret(secretName)
+	chunks := chunkPayload(payload, defaultChunkSize)
+	existingSecrets, err := c.getSecrets()
 	if err != nil {
-		if !k8serrors.IsNotFound(err) {
+		return err
+	}
+
+	for idx, data := range chunks {
+		secretName, err := c.createSecretName(idx)
+		if err != nil {
 			return err
 		}
 
-		secret = &unstructured.Unstructured{
-			Object: map[string]interface{}{
-				"metadata": metav1.ObjectMeta{
-					Name:        secretName,
-					Namespace:   c.namespace,
-					Labels:      c.getLabels(),
-					Annotations: map[string]string{"encoding": "gzip"},
+		secret, err := c.getSecret(secretName)
+		if err != nil {
+			if !k8serrors.IsNotFound(err) {
+				return err
+			}
+			secret = &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"metadata": metav1.ObjectMeta{
+						Name:        secretName,
+						Namespace:   c.namespace,
+						Labels:      c.getLabels(),
+						Annotations: map[string]string{"encoding": "gzip"},
+					},
 				},
-			},
+			}
+			secret, err = c.kubernetesSecretClient.Create(ctx, secret, metav1.CreateOptions{})
```

Comment:
AFAIK the Create call translates into a POST HTTP method. How does this behave when the Secret for the chunk already exists?

Reply:
The logic above this checks if the secret already exists or not and only calls create if it does not already exist. We could however reformulate this as a single patch call instead. This is logic that was already present for the single secret backend, it has just been shuffled around in this PR.